portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -93,26 +113,72 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String content = request.getParameter("content");
     Date date = new Date();
     long timestamp = System.currentTimeMillis();
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageURL = getUploadedFileUrl(request, "imageURL");
 
     // Creates Entity object.
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("username", username);
     commentEntity.setProperty("content", content);
     commentEntity.setProperty("date", date);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("imageURL", imageURL);
 
     // Connects to the datastore and inserts the entity.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     // Redirect back to the HTML forum page.
-    response.sendRedirect("/blog.html");
+    response.sendRedirect("/forum.jsp");
   }
 
+  // Helper functions/routines:
+
+  // Creates a comment object from the inputs.
   private Comment createComment(Entity entity) {
     String username = entity.getProperty("username").toString();
     String content = entity.getProperty("content").toString();
+    Object imageURLObject = entity.getProperty("imageURL");
+    String imageURL = null;
+    if (imageURLObject != null) imageURL = imageURLObject.toString();
     Date date = (Date) entity.getProperty("date");
-    return (new Comment(username, content, date));
+    return (new Comment(username, content, imageURL, date));
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
if you're intending to do this later, you should put "TODO:" at the start of the comment.  If you don't want to do it later, you should just remove it.